pandas/core/algorithms.py

```diff
@@ -429,7 +429,14 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
         # handle categoricals
         return comps.isin(values)  # type: ignore
 
-    comps, dtype = _ensure_data(comps)
+    if hasattr(comps, "dtype") and hasattr(comps.dtype, "name"):
+        if comps.dtype.name == "int64":
+            comps, dtype = _ensure_data(comps, dtype=object)
+        else:
+            comps, dtype = _ensure_data(comps)
+    else:
+        comps, dtype = _ensure_data(comps)
+
     values, _ = _ensure_data(values, dtype=dtype)
```

Comment:
The line above should be fine; I think this is the line where things actually go awry.

This has been in place at least since #15929 maybe further back. Not sure what breaks if you don't align the dtype here but yea I think should leave the preceding line as is

Reply:
My first try was just take out the dtype when calling the _ensure_data but this break others parts of the code when testing cause of that i needed to do some conditions to leave the dtype Blanked just in some cases. I couldn’t get what do you think I should do.

Reply:
Do you think it is right to consider “1” equals 1? For me that looks pretty bad